plugins/modules/cloudflare_dns.py

```diff
@@ -407,7 +428,7 @@ def __init__(self, module):
         if (self.type in ['CNAME', 'NS', 'MX', 'SRV']) and (self.value is not None):
             self.value = self.value.rstrip('.').lower()
 
-        if (self.type == 'AAAA') and (self.value is not None):
+        if (self.type in ['AAAA', 'CAA']) and (self.value is not None):
             self.value = self.value.lower()
```

Comment:
Sorry I only noticed this now, but: why make the value lower-case? Parts of the CAA record could be case-insensitive, for example the ACME account ID (https://www.rfc-editor.org/rfc/rfc8657#section-3) - I cannot find any indication in the RFC that its case does not matter (I also didn't search very long for it, but maybe you found some evidence that casing does not matter?).

Reply:
yes and no, 
They are Uri's, so there are normalization rules ;)
[RFC3986 6.2.2.1](https://www.rfc-editor.org/rfc/rfc3986#section-6.2.2.1).  Case Normalization

Reply:
As long as the CAA RFCs don't say that casing doesn't matter I would not apply any normalization without good reason. So far I haven't seen a good reason why it should be done here.